app/common/src/services/Backend.ts

```diff
@@ -793,17 +799,31 @@ export interface LChColor {
   readonly alpha?: number | undefined
 }
 
+/** Type used when creating api key credential. */
+export interface CreateApiKeyRequestBody {
+  readonly name: string
+  readonly description: string
+  readonly expiresIn: ApiKeyExpiresIn
+}
+
+/** Api key credential. */
 export interface ApiKey {
   readonly id: ApiKeyId
+  // Field populated only once after creation.
+  readonly secretId: string | null
   readonly name: string
   readonly description: string
   readonly createdAt: dateTime.Rfc3339DateTime
   readonly lastUsedAt: dateTime.Rfc3339DateTime | null
+  readonly expiresAt: dateTime.Rfc3339DateTime | null
 }
 
-export interface CreateApiKeyRequestBody {
-  readonly name: string
-  readonly description: string
+/** Possible types of lifetime span for api key credentials. */
+export enum ApiKeyExpiresIn {
+  Week = 'Week',
+  Month = 'Month',
+  Year = 'Year',
+  Indefinetly = 'Indefinetly',
```

Comment:
@PabloBuchu `Indefinitely` is the correct spelling (but maybe `Indefinite` is more appropriate, given that the others aren't `Yearly` (for example))